std/assembly/math.ts

```diff
@@ -992,6 +992,11 @@ export namespace NativeMath {
     }
   }
 
+  @inline
+  export function signbit(x: f64): bool {
+    return reinterpret<u64>(x) >>> 63 != 0;
```

Comment:
> 1. If n is NaN, the result is false.

Does this hold here?

Reply:
you mean sNaN and qNaN? Hmm, I investigate how Rust handle this and it seems it do similar:
https://github.com/rust-lang/rust/pull/42431/files#diff-c371f0bfa18fe1918ce4cdc6f3026e71R224
GO: https://golang.org/src/math/signbit.go

And etc. I think until spec not stabilized this approach is most suitable

Reply:
I saw this. Actually `signbit` mean we extract MSB sign bit as is. Isn't?
cc @jfbastien

Reply:
Some approach for [`__builtin_signbit` in GCC](https://github.com/gcc-mirror/gcc/blob/master/gcc/builtins.c#L5868)

Reply:
The right shift is fine, really just thinking about the `false` for NaN rule, i.e.

```ts
return x == x && (reinterpret<u64>(x) >>> 63) != 0;
```

Reply:
I found also this discussion about signed NaN in Rust:
https://github.com/rust-lang/rust/issues/42425#issuecomment-306053495

And they switched to current behaviour.

Reply:
I do not quite understand the comparison to other languages, when the goal is to implement the JS spec?

Reply:
Because this not yet stabilized spec (Stage 1).

Reply:
Yeah, but shouldn't we go with what's in stage 1 for now, until we know better? A minimal constant-time implementation could be, for now:

```ts
function signbit64(x: f64): bool {
  return <bool>(reinterpret<u64>(x) >>> 63) & (x == x);
}

function signbit32(x: f32): bool {
  return <bool>(reinterpret<u32>(x) >>> 31) & (x == x);
}

```

Reply:
Open issue in [proposal repo](https://github.com/tc39/proposal-Math.signbit/issues/1)

Reply:
"In some implementations, external code might be able to detect a difference between various Not-a-Number values, but such behaviour is implementation-dependent; to ECMAScript code, all NaN values are indistinguishable from each other." (https://tc39.github.io/ecma262/#sec-ecmascript-language-types-number-type)

@dcodeIO it seems you are right =)